### PR TITLE
Automated builds for gnu/Linux, macOS and Windows

### PR DESCRIPTION
**Pull request by arky**
Contributing a github action to do automated builds for multiple OS and platforms.

### CODE REVIEW

**Comment by arky** [comment]
@vinayak-mehta Please kindly do review this PR before it turns stale. Thanks!

File: .github/workflows/default.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  build:
+    runs-on: ${{ matrix.config.os }}
+
+    strategy:
+      matrix:
+        config:
+          - { os: macos-latest, arch: x64, name: macos-latest }
+          - { os: ubuntu-latest, arch: x64, name: ubuntu-latest }
+          - { os: windows-latest, arch: x64, name: windows-latest }
+          - { os: windows-latest, arch: ia32, name: windows-latest-32 }
```
Comment (vinayak-mehta) [inline .github/workflows/default.yml:19]: @arky Will `windows-latest` build it for windows 10? Is that build going to work on windows 7 which #99 is about? I don't have a windows 7 / 10 to test out the build.

Comment (vinayak-mehta) [reply .github/workflows/default.yml:19]: But I guess we can run this workflow and build it anyway.

**Comment by vinayak-mehta** [comment]
Changed the trigger to manual for now, we can configure it to run whenever a new tag is pushed later.

File: .github/workflows/default.yml (hunk 1 above)
Comment (arky) [reply .github/workflows/default.yml:16]: @vinayak-mehta According to Github Actions documentation the default windows-latest OS is Windows Server 2019 https://docs.github.com/en/actions/reference/virtual-environments-for-github-hosted-runners

The workflow with generate an artifact with '-ia32' suffix that could be tested on Windows 32bit environments. That will help us resolve the #99.

**Pull request merged**